WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift

```diff
@@ -183,12 +183,16 @@ fileprivate extension CardPresentPaymentEvent {
 }
 
 #if DEBUG
+import class WooFoundation.MockAnalyticsPreview
+import class WooFoundation.MockAnalyticsProviderPreview
+
 #Preview {
     let totalsVM = TotalsViewModel(orderService: POSOrderPreviewService(),
                                    cardPresentPaymentService: CardPresentPaymentPreviewService(),
                                    currencyFormatter: .init(currencySettings: .init()),
                                    paymentState: .acceptingCard)
-    let cartVM = CartViewModel()
+    let analytics = MockAnalyticsPreview(userHasOptedIn: true, analyticsProvider: MockAnalyticsProviderPreview())
```

Comment:
I'm not excited about repeating the initialization code but I see the value of injecting the Analytics protocol and not polluting POS with additional dependencies.

Could we define this `MockAnalyticsPreview` within POS and Woo target so it could be just injected with `MockAnalyticsPreview()`, without passing anything additional?

And in testing, just use separate mocked woo analytics. WDYT?

Reply:
I'm not sure if this is what you meant, let me know if it isn't:

If we just pass default values to the preview init we can clean up redudant initialization ( done on https://github.com/woocommerce/woocommerce-ios/pull/13672/commits/02467c08affcb65af0629a330a39688ee992d8fd ). Since the Preview version it's in WooFoundation, we just use this for previews, and for tests we'll keep using `WooAnalytics` and `MockAnalyticsProvider`.

I'm pretty sure we can still clean up a bunch of these dependencies in Previews, some are hideous at the moment, something to take a look later on when we have more time 😂 

On a different note: Confirmed in slack that we should register all events, so done it here:  https://github.com/Automattic/tracks-events-registration/pull/2597 👍

Reply:
Thanks. Yes, this is what I meant 👍